File: clients/py/queue_send.py
import json
import time
import uuid
import redis
from faker import Faker

fake = Faker()
host = 'localhost'
write_port = 6379
redis_password = 'data_store_password'
queue_name = 'tasks'

try:
    r = redis.Redis(host=host, port=write_port, db=0, password=redis_password)
    while True:
        try:
            id = uuid.uuid4()
            r.rpush(f'queues:{queue_name}', json.dumps({'name': fake.name(), 'id': str(id)}))
            print('.', end='', flush=True)
        except redis.ConnectionError:
            print('connection error')
            time.sleep(2)
            r = redis.Redis(host=host, port=write_port, db=0, password=redis_password)
        time.sleep(.1)
except KeyboardInterrupt:
    pass

File: clients/js/queueSend.js
import { createClient, commandOptions } from 'redis';
import { faker } from '@faker-js/faker';

const redisPassword = 'data_store_password';
const queueName = 'tasks';

(async () => {
  const client = createClient({password: redisPassword});
  client.on('error', (err) => console.log('Redis Client Error', err));

  await client.connect();

  setInterval(async () => {
    const user = {
      id: faker.string.uuid(),
      name: faker.person.fullName()
    };      
    const rPushPromise = client.rPush(
        commandOptions({ isolated: true }),
        `queues:${queueName}`,
        JSON.stringify(user)
    );
    const task = await rPushPromise;
    console.log(`${task} tasks in queue`);
  }, 100);
})();
